File: src/console/command/parameter/def/TemplateTypeParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\CommandParameter;
use pocketcloud\cloud\console\command\parameter\exception\ArgumentParseException;
use pocketcloud\cloud\template\TemplateType;

readonly class TemplateTypeParameter extends CommandParameter {

    public function parseValue(string $input): TemplateType {
        if (($templateType = TemplateType::get($input)) !== null) return $templateType;
        throw new ArgumentParseException();
    }

    public function onTabCompleteMatch(string $currentArg): array {
        return array_filter(array_keys(TemplateType::getAll()), fn(string $template) => str_contains(strtolower($template), $currentArg));
    }

    public function getType(): string {
        return "template_type";
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/parameter/def/TemplateTypeParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.api.template.TemplateType;
import de.pocketcloud.cloud.console.command.exception.ArgumentParseException;
import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;

import java.util.List;

public final class TemplateTypeParameter extends BaseCommandParameter {

    public TemplateTypeParameter(String name, boolean optional) {
        super(name, optional);
    }

    public TemplateTypeParameter(String name, boolean optional, String customErrorMessage) {
        super(name, optional, customErrorMessage);
    }

    @Override
    public Object parseValue(String input) throws ArgumentParseException {
        try {
            return TemplateType.valueOf(input);
        } catch (IllegalArgumentException _) {
            throw new ArgumentParseException("TemplateType not found");
        }
    }

    @Override
    public List<String> onTabCompleteMatch(String currentArg) {
        return List.of(TemplateType.SERVER.name(), TemplateType.PROXY.name());
    }

    @Override
    public String getType() {
        return "template_type";
    }
}